appinventor/codelabs/mole-mash/index.html

```diff
@@ -0,0 +1,242 @@
+        <p>From the <b>Math</b> drawer, drag a <i>multiplication</i> block into the <code>set Mole.X to</code> block.</p>
+        <img src="img/movemole2.png">
+        <p>From the <b>Math</b> drawer, drag a <code>random fraction</code> block into the first multiplication plug.</p>
+        <img src="img/movemole3.png">
+        <p>From the <b>Math</b> drawer, drag a <i>subtraction</i> block into the second multiplication plug.</p>
+        <img src="img/movemole4.png">
+        <p>From the <b>MyCanvas</b> drawer, drag a <code>MyCanvas.Width</code> block into the first subtraction plug.</p>
+        <img src="img/movemole5.png">
+        <p>From the <b>Mole</b> drawer, drag a <code>Mole.Width</code> block into the second subtraction plug.</p>
+        <img src="img/movemole6.png">
+        <h2>Challenges</h2>
+        <ol>
+          <li>Finish the MoveMole procedure by adding blocks to change the position of the Mole's y position.</li>
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Define a procedure to update the score">
+        <p>Our second prodecure will update the score of the game.</p>
+        <p>First, we need a variable to keep track of the score. Define a global variable named "score", and set the initial value to 0. This is what your block should look like:</p>
+        <img src="img/score_variable.png" style="max-width: 300px">
+        <p>Next, define a procedure named "UpdateScore". This procedure will change the text of the the label <b>ScoreLabel</b> to show the score of the game.</p>
+        <h2>Challenges</h2>
+        <ol>
+          <li>Try to code this procedure by yourself.</li>
+          <p>Hint: The completed UpdateScore procedure should look something like this:</p>
+          <img src="img/updatescore.png" style="max-width: 400px">
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add a Timer">
+        <p>The next step is to keep the mole moving. Every time MoleTimer goes off, the procedure MoveMole is called, meaning that the Mole has moved to a new, random position.</p>
+        <p>From the <b>MoleTimer</b> drawer, drag out a <code>when MoleTimer.Timer</code> block, and from the <b>Procedures</b> drawer, drag a <code>call MoveMole</code> block into the <code>when MoleTimer.Timer</code> block.</p>
+        <img src="img/moletimer.png" style="max-width: 250px">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add a Mole Touch handler">
+        <p>The program should increment the score each time the mole is touched. Sprites, like canvases, respond to touch events.</p>
+        <p>Every time the Mole is touched, your program will increment the score, call the UpdateScore prodecure to show the new score, make the phone vibrate for 100 milliseconds, and call the MoveMole procedure so the mole moves right away.</p>
+        <h2>Challenges</h2>
+          <ol>
+            <li>Try to code this procedure by yourself.</li>
```

Comment:
How about instead, we include screenshots of the blocks for each item in the list above and get rid of the answer screenshot here.

Feel free to move the paragraph above into the challenge description.

Typeblocking should not be a separate challenge. Maybe just add a note as part of the first (and only) challenge.